app/utils/verified-builds.tsx

```diff
@@ -39,111 +55,171 @@ export function useVerifiedProgramRegistry({
     options?: { suspense: boolean };
     programData?: ProgramDataAccountInfo;
 }) {
-    const { url: clusterUrl, cluster: cluster } = useCluster();
-    const connection = new Connection(clusterUrl);
-
     const {
         data: registryData,
         error: registryError,
         isLoading: isRegistryLoading,
     } = useSWRImmutable(
         `${programId.toBase58()}`,
         async (programId: string) => {
-            const response = await fetch(`${OSEC_REGISTRY_URL}/status/${programId}`);
+            const response = await fetch(`${OSEC_REGISTRY_URL}/status-all/${programId}`);
 
-            return response.json();
+            return response.json() as Promise<OsecInfo[]>;
         },
         { suspense: options?.suspense }
     );
 
-    if (programData && registryData) {
-        const hash = hashProgramData(programData);
-        registryData.verification_status =
-            hash === registryData['on_chain_hash'] ? VerificationStatus.Verified : VerificationStatus.NotVerified;
+    if (!programData || !registryData) {
+        return { data: null, error: registryError, isLoading: isRegistryLoading };
     }
 
+    // Only trust entries that are verified and signed by a trusted signer or the program authority
+    const trustedEntries = registryData.filter(
+        entry => (TRUSTED_SIGNERS[entry.signer] || entry.signer === programAuthority?.toBase58()) && entry.is_verified
+    );
+
+    // Update the verification status of the trusted entries based on the on-chain hash
+    const hash = hashProgramData(programData);
+    trustedEntries.forEach(entry => {
+        entry.is_verified = hash === entry['on_chain_hash'];
+    });
+
+    return { data: trustedEntries, isLoading: isRegistryLoading };
+}
+
+// Method to fetch verified build information for a given program
+// Returns the first verified entry that is signed by the program authority or a trusted signer
+export function useVerifiedProgram({
+    programId,
+    programAuthority,
+    options,
+    programData,
+}: {
+    programId: PublicKey;
+    programAuthority: PublicKey | null;
+    options?: { suspense: boolean };
+    programData?: ProgramDataAccountInfo;
+}) {
+    const { data: registryData } = useVerifiedProgramRegistry({
+        options,
+        programAuthority,
+        programData,
+        programId,
+    });
+
+    const mappedBySigner: Record<string, OsecInfo> = {};
+
+    // Map the registryData by signer in order to enforce hierarchy of trust
+    registryData?.forEach(entry => {
+        mappedBySigner[entry.signer] = entry;
+    });
+
+    // Get the program authority's entry first, then the trusted signers
```

Comment:
Would it make sense to get these already sorted from the API? Would make this easier to implement for other explorers

Reply:
I think it's fair to ask explorers to sort the results because each one is responsible for deciding what "verified" means to them.